File: plugin/testDataSwift/shared/basics/EscapingLambdas.shared.swift
//Package: com.test
//Converted using Khrysalis2

import Foundation



public class HasLambda {
    
    public var action:  () -> Void
    
    
    public func invoke() -> Void {
        action()
    }
    
    public init(action: @escaping () -> Void = { () in 
    }) {
        self.action = action
    }
    convenience public init(_ action: @escaping () -> Void) {
        self.init(action: action)
    }
}
 
 
public var globalThing: () -> Void = { () in 
}

public func doThing(action: @escaping () -> Void) -> Void {
    globalThing = action
}
 
 

public func main() -> Void {
    HasLambda{ () in 
        print("Hello world!")
    }.invoke()
    doThing{ () in 
        print("Hello world 2!")
    }
    globalThing()
}
 

File: plugin/testDataOutputTs/shared/basics/EscapingLambdas.shared.ts
//Package: com.test
//Converted using Khrysalis2


class HasLambda {
    action:  () => void;
    
    public constructor(action:  () => void = () => {
    }) {
        this.action = action;
    }
    
    invoke(): void {
        this.action()
    }
}

 

let globalThing : () => void = () => {
} 
;
export function doThing(action:  () => void): void {
    globalThing = action
}

 

export function main(): void {
    (new HasLambda(() => {console.log("Hello world!")
    })).invoke();
    doThing(() => {console.log("Hello world 2!")
    });
    globalThing()
}

